src/network/message_network.rs

```diff
@@ -78,6 +84,66 @@ impl_consensus_encoding!(VersionMessage, version, services, timestamp,
                          receiver, sender, nonce,
                          user_agent, start_height, relay);
 
+#[repr(u8)]
```

Comment:
Why the repr?

Reply:
to be able to convert the enum into u8 with as u8

Reply:
I'll try with 1.22 but with newer rust versions it works just fine:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=88287b6224e3e4b5bb215ef26fe323e7

Reply:
Just tried with 1.22.0 and works.
the repr is about memory representation, not about casting.
repr's are only if you pass C ABI or you transmute to a different one.

Reply:
sounds like a serious problem

Reply:
Why a problem?
I just don't think it's necessary

Reply:
I think what he's saying is that `as u8` also works without the `repr`. `repr` is used when you are using the type in a C API. So I agree we don't need it here.